### ISSUE TITLE
Handle device geolocation enabling in Location-kit

### ISSUE THREAD

**Issue by MaratEldarov** [HMS-Core/hms-cordova-plugin#20]
**Description**
I need to implement location obtain at the start of the cordova application. I get permissions with `HMSFusedLocation.requestPermission`, after that I call `HMSFusedLocation.checkLocationSettings` and if geolocation on device was enable - I'm not getting error and can get location via `HMSFusedLocation.getLastLocation`. But if geolocation wasn't enable - I got pop-up with proposal of enable geolocation and at the same time HMSFusedLocation.getLastLocation works and return error because of `HMSFusedLocation.checkLocationSettings` resolved.

**Expected behavior**
I think right behavior would be that if geolocation on device is disable - `HMSFusedLocation.checkLocationSettings` not resolving until user select to enable (or not) geolocation on device. 

**Current behavior**
`HMSFusedLocation.checkLocationSettings` resolving before user select to enable (or not) geolocation on device and so next method `HMSFusedLocation.getLastLocation` not getting any geolocation data.

**Screenshots**
If applicable, add screenshots to help explain your issue

**Logs**
Post logs here or paste them to [Ghostbin](https://ghostbin.co) and insert the link here.

**Environment**
 - Platform: Cordova
 - Kit: Location-kit
 - 5.0.0-301
 - Android 11

**Other**
```
const requestLocationPermissionsHuawei = async () => {
  if (window.HMSFusedLocation) {
    try {
      const result = await window.HMSFusedLocation.requestPermission();
      return result;
    } catch (error) {
      console.error(error);
    }
  }
};

const locationRequest = () => ({
  id: `locationRequest${Math.random() * 10000}`,
  priority: window.HMSFusedLocation.PriorityConstants.PRIORITY_HIGH_ACCURACY,
  interval: 3,
  numUpdates: 10,
  fastestInterval: 1000.0,
  expirationTime: 200000.0,
  expirationTimeDuration: 200000.0,
  smallestDisplacement: 0.0,
  maxWaitTime: 2000000.0,
  needAddress: true,
  language: 'fr',
  countryCode: 'ru',
});

const locationSettingsRequest = () => ({
  alwaysShow: false,
  needBle: true,
  locationRequests: [locationRequest()],
});

const geolocationGetCurrentPositionHuawei = async () => {
  try {
    const { granted } = await requestLocationPermissionsHuawei();
    if (granted) {
      const locationSettings = locationSettingsRequest();
      const locationSettingsResult = await window.HMSFusedLocation.checkLocationSettings(locationSettings);
      const requestLocationUpdatesResult = await window.HMSFusedLocation.requestLocationUpdates(locationRequest());
      const lastLocationResult = await window.HMSFusedLocation.getLastLocation();
      const { latitude = null, longitude = null } = lastLocationResult;
      return {
        coords: {
          latitude,
          longitude,
        },
      };
    }
  } catch (error) {
    console.error(error);
  }
};

```

**Comment by Zeevac** [comment]
Hello. I will help you but first of all, let's clarify some things. When you say 'enable geolocation', did you mean giving location permission? Second question, what is your cordova cli and cordova-android version?

**Comment by MaratEldarov** [comment]
Hi @Zeevac ! I mean enabling geolocation on device:

![eQbm2YYhKcw](https://user-images.githubusercontent.com/35111740/109931866-d8ff5f80-7ce2-11eb-9d77-9263473fe030.jpg)
 
In my case it disable when I call `HMSFusedLocation.checkLocationSettings`, and after i call it - i got proposal of enable geolocation on device.

Cordova version 8.1.2
Cordova android 8.0.0

**Comment by Zeevac** [comment]
Hello @MaratEldarov. I inspected the issue. This issue is fixed on new release. Please wait for new release.

**Comment by MaratEldarov** [comment]
@Zeevac  that's good! Any info about when it will be released?

**Comment by aktug** [comment]
Released v5.1.0-303

https://www.npmjs.com/package/@hmscore/cordova-plugin-hms-location
https://www.npmjs.com/package/@hmscore/ionic-native-hms-location
